ci/kind/kind-upgrade.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+LATEST_KIND_VERSION=$(curl -sSL https://api.github.com/repos/kubernetes-sigs/kind/releases/latest | jq -r '.tag_name' | awk '{print substr($1, 2)}')
+
+# on linux
+function upgrade_kind {
+    arch_name="$(uname -m)"
+    binary_name=""
+    case "$arch_name" in
+     "x86_64")
+      binary_name="kind-linux-amd64"
+      ;;
+      "aarch64")
+      binary_name="kind-linux-arm64"
+      ;;
+      *)
+      echoerr "Unsupported platform $arch_name"
+      exit 1
+      ;;
+    esac
+
+    set +e
+    curl -Lo ./kind https://kind.sigs.k8s.io/dl/v$LATEST_KIND_VERSION/$binary_name
+    chmod +x ./kind
+    sudo mv ./kind /usr/local/bin/kind
+    if [[ $? -ne 0 ]]; then
+        echoerr "Failed to update kind version v$LATEST_KIND_VERSION"
+    else
+        echo "Kind version v$LATEST_KIND_VERSION updated successfully"   
+    fi
+    set -e
+}
+
+function compare_kind {
+    kind_version=$(kind version | awk '{print substr($2, 2)}') # strip leading 'v'
+    if [[ $(echo "$kind_version $LATEST_KIND_VERSION" | tr " " "\n" | sort --version-sort | head -n 1) != "$LATEST_KIND_VERSION" ]]; then
```

Comment:
I don't understand why this is not simply `[[ "$kind_version" != "$LATEST_KIND_VERSION" ]]`. Could you explain?

Reply:
my bad, we can directly compare it no need to sort it. thanks!